Bug: sstore

https://github.com/kkrt-labs/keth/actions/runs/12898622580/job/35966030584

```
___________________________ TestStorage.test_sstore ____________________________
[gw14] linux -- Python 3.10.16 /home/runner/work/keth/keth/.venv/bin/python

self = <test_storage.TestStorage object at 0x7ff9639dd0c0>
cairo_run = <function _run_rust_vm.<locals>._run at 0x7ff932234e50>

    @given(evm=evm_with_accessed_storage_keys())
>   def test_sstore(self, cairo_run, evm: Evm):

tests/ethereum/cancun/vm/instructions/test_storage.py:50: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <test_storage.TestStorage object at 0x7ff9639dd0c0>
cairo_run = <function _run_rust_vm.<locals>._run at 0x7ff932234e50>
evm = Evm(pc=Uint(1), stack=[U256(64), U256(768), U256(11), U256(9466), U256(42587), U256(57369), U256(16220), U256(46656), ...x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00')})

    @given(evm=evm_with_accessed_storage_keys())
    def test_sstore(self, cairo_run, evm: Evm):
        try:
            cairo_evm = cairo_run("sstore", evm)
        except Exception as cairo_error:
            with strict_raises(type(cairo_error)):
                sstore(evm)
            return
        sstore(evm)
>       assert evm == cairo_evm
E       assert gas_left field mismatch:
E           left:  17372117
E           right: 17374316
E       Falsifying example: test_sstore(
E           self=<test_storage.TestStorage object at 0x7ff9639dd0c0>,
E           cairo_run=_run,
E           evm=Evm(pc=Uint(0),
E            stack=[U256(64),
E             U256(768),
E             U256(11),
E             U256(9466),
E             U256(42587),
E             U256(57369),
E             U256(16220),
E             U256(46656),
E             U256(159),
E             U256(14645),
E             U256(165),
E             U256(44904),
E             U256(0),
E             U256(35224)],
E            memory=Memory(b''),
E            code=b'',
E            gas_left=Uint(17374317),
E            env=Environment(caller=b'j0\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
E             block_hashes=[b'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX',
E              b'\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce\xce',
E              b'\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d\x8d',
E              b'\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85\x85',
E              b'\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d\x9d',
E              b'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
E              b'\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14\x14',
E              b'\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1\xe1',
E              b'\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f\x1f',
E              b'dddddddddddddddddddddddddddddddd',
E              b'22222222222222222222222222222222',
E              b'\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4\xc4',
E              b'\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e\x9e',
E              b'GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG',
E              b'\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff',
E              b'\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82\x82',
E              b'EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE',
E              b'\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8\xb8',
E              b'\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7\xd7',
E              b'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
E              b'88888888888888888888888888888888',
E              b'\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf',
E              b'\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b\x9b',
E              b'--------------------------------',
E              b'================================'],
E             origin=b'\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
E             coinbase=b'\x93\xbe\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
E             number=Uint(25),
E             base_fee_per_gas=Uint(U64(14_687_317_050_282_141_798)),
E             gas_limit=Uint(U64(39364)),
E             gas_price=Uint(U64(19027)),
E             time=U256(81),
E             prev_randao=b'\xcb\xd0\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
E             state=State(_main_trie=Trie(secured=True,
E               default=None,
E               _data={b'G\x1a\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': Account(nonce=Uint(
E                     U64(23),
E                 ),
E                 balance=U256(29),
E                 code=b'\x8e\xa6\xc2\xf24DB8\xaed\xb1\xf1\xc0\xaa\x85\xa9\xc3\x01'),
E                b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': Account(nonce=Uint(
E                     U64(50292),
E                 ), balance=U256(28525), code=b'\xd1\xe5U'),
E                b'\x18\xbe\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': Account(nonce=Uint(
E                     U64(56768),
E                 ), balance=U256(3_458_722_244), code=b''),
E                b'\xde\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': Account(nonce=Uint(
E                     U64(24573),
E                 ), balance=U256(18603), code=b'>k0\xec\xf0\xb1'),
E                b'5\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': Account(nonce=Uint(
E                     U64(39895),
E                 ), balance=U256(45447), code=b'C$\xf2')}),
E              _storage_tries={},
E              _snapshots=[(Trie(secured=True,
E                 default=None,
E                 _data={b'G\x1a\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': Account(nonce=Uint(
E                       U64(23),
E                   ),
E                   balance=U256(29),
E                   code=b'\x8e\xa6\xc2\xf24DB8\xaed\xb1\xf1\xc0\xaa\x85\xa9\xc3\x01'),
E                  b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': Account(nonce=Uint(
E                       U64(50292),
E                   ), balance=U256(28525), code=b'\xd1\xe5U'),
E                  b'\x18\xbe\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': Account(nonce=Uint(
E                       U64(56768),
E                   ), balance=U256(3_458_722_244), code=b''),
E                  b'\xde\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': Account(nonce=Uint(
E                       U64(24573),
E                   ), balance=U256(18603), code=b'>k0\xec\xf0\xb1'),
E                  b'5\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': Account(nonce=Uint(
E                       U64(39895),
E                   ), balance=U256(45447), code=b'C$\xf2')}),
E                {})],
E              created_accounts={b'\n\xac\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
E               b'N\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
E               b'\x9a\x8e\xd8vt\xe49\x99h;/\x83\xb1\xaf=xr\xb9[\xf3',
E               b'\xb4\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
E               b'\xb5\xee\xf3\x18|I]\x82\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
E               b'\xba4\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'}),
E             chain_id=U64(18370),
E             excess_blob_gas=U64(6_376_361_187_440_289_249),
E             blob_versioned_hashes=(b'\xda\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
E              b'\xfc\x93\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'),
E             transient_storage=TransientStorage(_tries={b'\x90*\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': Trie(secured=True,
E                default=U256(0),
E                _data={b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     57,
E                 ),
E                 b',\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     2021,
E                 )}),
E               b'\xb3\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': Trie(secured=False,
E                default=U256(0),
E                _data={b'\x8e\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     34,
E                 ),
E                 b'\xd8\xc1\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     16141,
E                 ),
E                 b'A\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     150,
E                 ),
E                 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     105,
E                 )}),
E               b'\x8c\xf6\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': Trie(secured=True,
E                default=U256(0),
E                _data={b'e\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     1,
E                 ),
E                 b'\x02\xf9\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     88_400_018_563_261_890_298_781_941_109_322_847_233_737_595_214_855_759_645_014_165_887_640_716_958_033,
E                 ),
E                 b'E\xba\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     5184,
E                 ),
E                 b'\xe6\xff\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     20115,
E                 )}),
E               b'VM>\r\xc8c\x9a\r\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': Trie(secured=True,
E                default=U256(0),
E                _data={b'~L\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     158,
E                 ),
E                 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     3574,
E                 ),
E                 b"'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00": U256(
E                     49444,
E                 ),
E                 b'\xfe\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff': U256(
E                     1,
E                 ),
E                 b'\xe1K\x85V\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     10576,
E                 ),
E                 b'I\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     195,
E                 )}),
E               b'\xca\xb1PK\x9c\xcc\xf1R\x08\xa7o\xd2"\xcf\x89\x04I\x12\xf3\x00': Trie(secured=True,
E                default=U256(0),
E                _data={b'\x08\xf2\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     91,
E                 ),
E                 b'=|\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     115_792_089_237_316_195_423_570_985_008_687_907_853_269_984_665_640_564_039_457_584_007_913_129_639_935,
E                 ),
E                 b'\t\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     5464,
E                 ),
E                 b'\xbe\xf3D\x9f\x0e_\xa0\xb5\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     16_571_996_716_842_295_939,
E                 ),
E                 b'`\xb73_!\xbao\x9eixD\xed\x88\x9a\xa8\t\xc7\xea\x01z\xa2\xeeIC\x8be\xf7\x1c\xed\x8d\r\xd5': U256(
E                     82_944_001_599_530_086_823_419_921_039_720_322_948_173_118_854_254_123_963_669_293_053_233_194_167_294,
E                 ),
E                 b'\xa8\x8f\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     28309,
E                 ),
E                 b'r[\x84\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     290,
E                 ),
E                 b'\xac\x8a\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     30103,
E                 ),
E                 b'Pu\xa5>\x0e\x1c\xcdW@\xa3\x89\xc23/\xa5_ \x18\x9d \xb6\x14\xe2\xfc\x9a\xf9R\x01-\xc59\t': U256(
E                     1_158_781_967,
E                 ),
E                 b'\xdb\xdc\xa6\xfe\xd1>\xa2\xc3?\x815\x1f\xe6]z\x0c\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     42071,
E                 ),
E                 b'\xeb\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     22022,
E                 ),
E                 b'\xa3\x05\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     63420,
E                 ),
E                 b'\xed\xa2\xf7\xd1\t\x9f@\xac\x99\xd4X\xf9\x04m\xb1\xf61\x91\x0c\xff\xba\xe1\x11\xe0G\xf0\x9c\x01\xb9\x11\xda\n': U256(
E                     61298,
E                 ),
E                 b"f\xfe\x81d\x16Z\x18q\x18<L\x97\xc4\xa1\xbe\x95\x15\xdc\x1a\xad\xd9\x85\x7f\xe0\x93'\xfa\x97\xf8\xb4\x86i": U256(
E                     50270,
E                 ),
2025-01-22 00:36:21,471 - INFO - Using Hypothesis profile: nightly
2025-01-22 02:12:21,288 - INFO - Controller worker: collecting tests to skip
E                 b'm\x18\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00': U256(
E                     210,
E                 )})},
E              _snapshots=[])),
E            valid_jump_destinations=set(),
E            logs=(),
E            refund_counter=0,
E            running=True,
E            message=Message(caller=b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
E             target=b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
E             current_target=b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
E             gas=Uint(0),
E             value=U256(0),
E             data=b'',
E             code_address=None,
E             code=b'',
E             depth=Uint(0),
E             should_transfer_value=False,
E             is_static=False,
E             accessed_addresses=set(),
E             accessed_storage_keys=set(),
E             parent_evm=None),
E            output=b'',
E            accounts_to_delete=set(),
E            touched_accounts=set(),
E            return_data=b'',
E            error=ethereum.exceptions.EthereumException(),
E            accessed_addresses=set(),
E            accessed_storage_keys={(b'x\xc6\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
E              b'\xe5\xfc\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'),
E             (b'y\x9f\xf8y\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
E              b'\x98\x89\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'),
E             (b'{\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
E              b'\x9c\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'),
E             (b'\xed+\xcb\x1b\xb2\xff>\xd4\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
E              b'\xbd\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00')}),
E       )
E       
E       You can reproduce this example by temporarily adding @reproduce_failure('6.123.17', b'AXicjZL9U5RVFMfPSU0XWRsYrNVoBIaZTYvxFQamWDjneRjYNGdFRSbExQBHMsIYbcC0xpcNycQcxpAVBEGIEnW2VGRHUWFLR9SEZETTZtcRR7P1BRxWiNjO0x/QdH+45849n/v9nnvuRQV+LlJgpA/pU4YqNMM1k/+HVzvf8DKcRrX4eU2xAqV7ASkReYyEiciRw6hAY4ZM7inICctlcUKyUIMcFy2xQTaOrkYCUFEXmk9TeNYHlM5whWEng42hmlZSCIOHplEOzWFwMeynZAY/wxZKYjjJcF2QWIYzDPsoiuJJJNvKzdD5a09Apv3+KgUqXfzOYlqkwNVO5KnJ4qYhBrH/HSka6BV6rbCssb1/nsqxR3IcT89+b/uuQxDXOi7Iz+7qW6rA7bNJcMX4wJXF1r855dAq05pZfz5xKLDvugKOkmwlst/i9IR3X9r2sN94esZ5R21sjwg0B0iTtmSaNxoGHh0Xu2NIC3EpDGS0FhbFH3VsnfnW6sq4u+s+6S2zI89zAie3p6Rver9lrS/XI/hNkSgfAVwCA8HmsaW/RHQXfDsh9emlqvkWx0XJDX6FKXp79gW96d005Blfy5EfAeV6FIf0Jo/vA5CtMooQ9lwvJ+iRiOEbDcjTuFxCyQy9YIVw6FhbH3XOXdLiCry7pPbO8PhDzdsfX1ZXNO0uvnD7gfHGIgGdSRySKNF/jxeWy3le8Jk8hqixfhDJKB6RqEn5/3uMiquaZpvv4ekWJDNDB2hl9E+gDOSN8f9DwY+k48np0tvjtdZJNepAmxn6/toVE7M4Yuu/FfTod3pDfbk7FPOjug34x3ldk327Vy3K21/gDLfObcnSmJasg68/O2Pa0eqtDm43N5W4U6/GuMm390Pb7IbecWtGpaRdTfzRHlSDt2UUMmpN/LKZ11eIQ5gu7qcoTB2yj9wJORFWbQizNsyc215an7jscugkU8N6i5oUVPUSWgI3ZN6bFr05oaPO1DXaeOs3BQ4ukxd5yGkvIo+rV8B3SlMLuBnUilVPkt1BHqc/cPfsQUf+2KH0a5XNiTW6Ll+dV4HHhRqX98Wx5xXDxnL357Ybh6femryn7YCrYsHbho8N772cs3lUfrprBbIhn6EbgBBozD+h+II4') as a decorator on your test case
```

--- a/cairo/ethereum/cancun/vm/instructions/storage.cairo
+++ b/cairo/ethereum/cancun/vm/instructions/storage.cairo
@@ -176,18 +176,19 @@ func sstore{
     let is_original_eq_current = U256__eq__(original_value, current_value);
     let is_current_eq_new = U256__eq__(current_value, new_value);
     let is_original_zero = U256__eq__(original_value, zero_u256);
-    if (is_original_eq_current.value != 0) {
-        if (is_current_eq_new.value == 0) {
-            if (is_original_zero.value != 0) {
-                tempvar gas_cost = gas_cost + GasConstants.GAS_STORAGE_SET;
-            } else {
-                tempvar gas_cost = gas_cost + (
-                    GasConstants.GAS_STORAGE_UPDATE - GasConstants.GAS_COLD_SLOAD
-                );
-            }
+    let is_original_eq_current_and_current_diff_new = is_original_eq_current.value * (
+        1 - is_current_eq_new.value
+    );
+    if (is_original_eq_current_and_current_diff_new != 0) {
+        if (is_original_zero.value != 0) {
+            tempvar gas_cost = gas_cost + GasConstants.GAS_STORAGE_SET;
+        } else {
+            tempvar gas_cost = gas_cost + (
+                GasConstants.GAS_STORAGE_UPDATE - GasConstants.GAS_COLD_SLOAD
+            );
         }
     } else {
-        tempvar gas_cost = GasConstants.GAS_WARM_ACCESS;
+        tempvar gas_cost = gas_cost + GasConstants.GAS_WARM_ACCESS;
     }
     let gas_cost = [ap - 1];
 

